zkabacus-crypto/src/states.rs

```diff
@@ -182,6 +191,11 @@ impl MerchantBalance {
         }
     }
 
+    /// Indicate if the balance is zero.
+    pub fn is_zero(&self) -> bool {
+        self.into_inner() == 0
```

Comment:
This is my lack of Rust understanding, but why not `self.0 == 0`?

Reply:
`self.0` is a `Balance`, so can't be directly compared to `0`. Maybe per your other comment, it would be better to implement `is_zero` for `Balance` and call it here.